pkg/cache/cluster.go

```diff
@@ -702,6 +741,51 @@ func (c *clusterCache) processApi(client dynamic.Interface, api kube.APIResource
 	return nil
 }
 
+// checkPermission runs a self subject access review to check if the controller has permissions to list the resource
+func (c *clusterCache) checkPermission(ctx context.Context, reviewInterface authType1.SelfSubjectAccessReviewInterface, api kube.APIResourceInfo) (keep bool, err error) {
+	sar := &authorizationv1.SelfSubjectAccessReview{
+		Spec: authorizationv1.SelfSubjectAccessReviewSpec{
+			ResourceAttributes: &authorizationv1.ResourceAttributes{
+				Namespace: "*",
+				Verb:      "list", // uses list verb to check for permissions
+				Resource:  api.GroupVersionResource.Resource,
+			},
+		},
+	}
+
+	switch {
+	// if manage whole cluster or resource is cluster level and cluster resources enabled
+	case len(c.namespaces) == 0 || !api.Meta.Namespaced && c.clusterResources:
+		resp, err := reviewInterface.Create(ctx, sar, metav1.CreateOptions{})
+		if err != nil {
+			return true, err
+		}
```

Comment:
actually it's an inconclusive response, we could say false too but as the request errored either of those values can be possible I chose true because instead of denying without confirmation the function responds with confirmation when error occurs. Though currently, it doesn't matter as the error will always take precedence.

Reply:
updated to false